lib/instrumentation/@elastic/elasticsearch.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+module.exports = function initialize(_agent, elastic, _moduleName, shim) {
+  shim.setDatastore(shim.ELASTICSEARCH)
+  shim.setParser(queryParser)
+
+  shim.recordQuery(elastic.Transport.prototype, 'request', function wrapQuery(shim, _, __, args) {
+    const ctx = this
+    return {
+      query: JSON.stringify(args?.[0]),
+      promise: true,
+      opaque: true,
+      inContext: function inContext() {
+        getConnection.call(ctx, shim)
+      }
+    }
+  })
+}
+
+function queryParser(params) {
+  params = JSON.parse(params)
+  const path = params.path.split('/')
+  const actions = {
+    GET: 'search',
+    PUT: 'create',
+    POST: 'create',
+    DELETE: 'delete',
+    HEAD: 'exists'
+  }
+  const defaultCollection = 'any'
```

Comment:
we aren't experts in elastic search and this is accessing deep properties at times. thoughts on wrapping all the parsing into a try/catch and logging and defaulting to something(not sure what to default to though 😬 ). the last thing i want this to do is crash for customers

Reply:
I'd be in favor of a try/catch. We've tested against only a handful of possible use cases. For the default... `any` for the collection and `unknown` for the operation?

Reply:
Added.